src/DynamicNLPModels.jl

```diff
@@ -1735,4 +2104,75 @@ function _init_similar(mat, dim1::Number, T=eltype(mat))
     return new_mat
 end
 
+
+function LinearAlgebra.mul!(y::Vector{T},
+    Jac::LQJacobianOperator{T, V, M},
+    x::Vector{T}
+) where {T, V <: AbstractVector{T}, M <: AbstractMatrix{T}}
+    fill!(y, zero(T))
+
+    J   = Jac.Jac
+    N   = Jac.N
+    nu  = Jac.nu
+    nc  = Jac.nc
+    nsc = Jac.nsc
+    nuc = Jac.nuc
+
+    J1Bx = Jac.J1Bx
+    J2Bx = Jac.J2Bx
+    J3Bx = Jac.J3Bx
+
+    for i in 1:N
+        sub_B1 = @view J[(1 + (i - 1) * nc):(i * nc), :]
+        sub_B2 = @view J[(1 + nc * N + (i - 1) * nsc):(nc * N + i * nsc), :]
+        sub_B3 = @view J[(1 + nc * N + nsc * N + (i - 1) * nuc):(nc * N + nsc * N + nuc * i), :]
+
+        for j in 1:(N - i + 1)
+            LinearAlgebra.mul!(J1Bx, sub_B1, view(x, (1 + (j - 1) * nu):(j * nu)))
+            LinearAlgebra.mul!(J2Bx, sub_B2, view(x, (1 + (j - 1) * nu):(j * nu)))
+            LinearAlgebra.mul!(J3Bx, sub_B3, view(x, (1 + (j - 1) * nu):(j * nu)))
+
+            view(y, (1 + nc * (j + i - 2)):(nc * (j + i - 1) )) .+= J1Bx
+            view(y, (1 + nc * N + nsc * (j + i - 2)):(nc * N + nsc * (j + i - 1))) .+= J2Bx
+            view(y, (1 + nc * N + nsc * N + nuc * (j + i- 2)):(nc * N + nsc * N + nuc * (j + i - 1))) .+= J3Bx
+        end
+    end
+end
+
+function LinearAlgebra.mul!(
+    y::Vector{T},
+    Jac::LinearOperators.AdjointLinearOperator{T, LQJacobianOperator{T, V, M}},
+    x::Vector{T}
+) where {T, V <: AbstractVector{T}, M <: AbstractMatrix{T}}
+    fill!(y, zero(T))
+
+    J   = Jac'.Jac
```

Comment:
oulala. I understand the syntax is correct, but I think it's a bit too heavy. Can we use a getter to access to the linear operator stored inside `Jac`?

Reply:
A function `get_operator` for instance, that returns a pointer to the operator `Jac` :)